flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOperationUtils.java

```diff
@@ -213,4 +211,40 @@ public static void addColumnFamilyOptionsToCloseLater(
 			throw new IOException("Failed to acquire shared cache resource for RocksDB", e);
 		}
 	}
+
+	@VisibleForTesting
+	static RocksDBSharedResources allocateRocksDBSharedResources(long size, double writeBufferRatio, double highPriorityPoolRatio) {
+		long calculatedCacheSize = calculateActualCacheSize(size, writeBufferRatio);
+		final Cache cache = createCache(calculatedCacheSize, highPriorityPoolRatio);
+		final WriteBufferManager wbm = new WriteBufferManager((long) (writeBufferRatio * size), cache);
```

Comment:
The `WriteBufferManager` size should also be calculated with the deduced formula: `2 * total_memory_size * write_buffer_ratio / (3 + write_buffer_ratio)`